src/memray/_memray/tracking_api.cpp

```diff
@@ -219,9 +275,47 @@ PythonStackTracker::setMostRecentFrameLineNumber(int lineno)
     }
 }
 
+void
+PythonStackTracker::reloadStackIfTrackerChanged()
```

Comment:
I am a bit concerned with this function. Is very easy to miss some situation when this needs to be called or to also overdo it. What about some alternative design: we add a getter/setter around the `d_stack` vector for the frames and we check the generation number then and act appropriately. This may have some downsides but I think that will be always correct, no?

Reply:
We discussed a few options offline for improving this, but we couldn't come up with anything better than adding a getter that handles reloading the stack when fetching the stack tracker. I've implemented that.